Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Math/Uuid.h>
+#include <AzCore/Memory/Memory.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzToolsFramework/ComponentMode/EditorBaseComponentMode.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+
+#include <Editor/Source/ComponentModes/Joints/JointsComponentModeCommon.h>
+#include <Editor/Source/ComponentModes/PhysXSubComponentModeBase.h>
+
+namespace AZ
+{
+    class EntityComponentIdPair;
+}
+
+namespace PhysX
+{
+    class JointsComponentMode final
+        : public AzToolsFramework::ComponentModeFramework::EditorBaseComponentMode
+    {
+    public:
+        AZ_CLASS_ALLOCATOR_DECL;
+
+        JointsComponentMode(const AZ::EntityComponentIdPair& entityComponentIdPair, AZ::Uuid componentType);
+        ~JointsComponentMode();
+
+        // EditorBaseComponentMode ...
+        void Refresh() override;
+        AZStd::vector<AzToolsFramework::ActionOverride> PopulateActionsImpl() override;
+        AZStd::vector<AzToolsFramework::ViewportUi::ClusterId> PopulateViewportUiImpl() override;
+    private:
+        enum class ClusterGroups
```

Comment:
suggestion: An API comment (or regular comment) for this (and what Groupw 1/2/3 correspond to) would be super useful